File: javascript/main2.mjs
// Run: node javascript/main3.mjs
function dangerous(array, index) {
  // Remove toString for The result is undefined!
  return array[index].toString(); // TypeError when index is OOB (undefined) otherwise its just undefined!
}
function foo(array, counter) {
  if (counter === 0) return dangerous(array, counter + 9137);
  return foo(array, counter - 1);
}
const array = new Array(1000).fill(0);
const result = foo(array, 6);
console.log("The result is", result);

File: scala/Main2.scala
object Main2 {
  def dangerous(array: Array[Int], index: Int): Int = array(index)

  def foo(array: Array[Int], counter: Int): Int =
    if (counter == 0) dangerous(array, counter + 9137)
    else foo(array, counter - 1)

  def main(args: Array[String]): Unit = {
    val a = Array.fill(1000)(0)
    val r = foo(a, 6)
    println(s"The result is $r")
  }
}
